### ISSUE TITLE
Registering Service Worker causes uncaught "SecurityError: The operation is insecure" when Firefox cookie settings is set to "Keep unitl Firefox is closed"

### ISSUE THREAD

**Issue by angelkenneth** [angular/angular#28373]
# 🐞 bug report

### Affected Package

@angular/service-worker

### Is this a regression?

Probubly not, after looking at the commit history of [module.ts](https://github.com/angular/angular/blob/master/packages/service-worker/src/module.ts) it seems the issue has been there ever since.

### Description

After creating a base Project and adding Service Worker then opening it on Firefox (mine is at version 64): it works normally.

But after going to Firefox's cookie setting and setting its cookie preference to "Keep unitl Firefox is closed", the browser then complains about `SecurityError`

Although the comment at [module.ts line 46](https://github.com/angular/angular/blob/master/packages/service-worker/src/module.ts#L46) expresses that not returning the promise will prevent the App from crashing.  SDKs like [Sentry](https://sentry.io) does catch and log them. 

I do extra custom things with Sentry, and skipping this particular exception is needed (since its not actually an exception).

Adding an noop `.catch(() => void 0)` to [module.ts line 47](https://github.com/angular/angular/blob/master/packages/service-worker/src/module.ts#L47) does quiet things down.

Will submit a PR with that code, just going to wait for a while to see if the community has some thoughts first.

## 🔬 Minimal Reproduction

1. Create the Project with Service Worker

   ```
   ng new test-project
   cd test-project
   ng add @angular/pwa
   ng build --prod
   ```

2. Serve files using `http-server` and open `http://localhost:8080` using a later version of Firefox:

   ```
   http-server dist/test-project/
   ```

3. Go to Firefox's cookie preference to "Keep unitl Firefox is closed"
4. Open DevTools and see the stack trace.

## 🔥 Exception or Error

<details>
  <summary>ERROR Error: "Uncaught (in promise): SecurityError: The operation is insecure</summary>

```
ERROR Error: "Uncaught (in promise): SecurityError: The operation is insecure
Ef/</<@http://localhost:8080/main.5f246923deb4ae88197d.js:1:189419
0TWp/</</c</e.prototype.invoke@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7102
onInvoke@http://localhost:8080/main.5f246923deb4ae88197d.js:1:66110
0TWp/</</c</e.prototype.invoke@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7029
0TWp/</</i</t.prototype.run@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:2288
I/<@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:13769
0TWp/</</c</e.prototype.invokeTask@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7777
onInvokeTask@http://localhost:8080/main.5f246923deb4ae88197d.js:1:66022
0TWp/</</c</e.prototype.invokeTask@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7690
0TWp/</</i</t.prototype.runTask@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:2951
g@http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:10048
	P http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:13020
	D http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:12118
	Z http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:12220
	invoke http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7102
	onInvoke http://localhost:8080/main.5f246923deb4ae88197d.js:1:66110
	invoke http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7029
	run http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:2288
	I http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:13769
	invokeTask http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7777
	onInvokeTask http://localhost:8080/main.5f246923deb4ae88197d.js:1:66022
	invokeTask http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:7690
	runTask http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:2951
	g http://localhost:8080/polyfills.f5ce8ecea387adbb1689.js:1:10048
```
</details>

## 🌍  Your Environment

<pre><code>
Angular CLI: 7.2.3
Node: 10.7.0
OS: linux x64
Angular: 7.2.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router, service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.12.3
@angular-devkit/build-angular     0.12.3
@angular-devkit/build-optimizer   0.12.3
@angular-devkit/build-webpack     0.12.3
@angular-devkit/core              7.2.3
@angular-devkit/schematics        7.2.3
@angular/cli                      7.2.3
@angular/pwa                      0.12.3
@ngtools/webpack                  7.2.3
@schematics/angular               7.2.3
@schematics/update                0.12.3
rxjs                              6.3.3
typescript                        3.2.4
webpack                           4.28.4
</code></pre>

**Comment by gkalpak** [comment]
Apparently, Fiefox does not allow registering a ServiceWorker with "Keep unitl Firefox is closed" for cookies. (This is not specific to Angular.)
I don't think it is a good idea to try and hide the errors (especially since they do not affect the app). The errors explains what went wrong (could not register the SW) and that it is due to user settings, so I feel this is fine to show in the console.

**Comment by manklu** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1429714

**Comment by angelkenneth** [comment]
@gkalpak i agree, the developer should be aware that the Service Worker did not install, even if the Browser supports.

Taking inspiration from [OneSignal's impletation](https://github.com/OneSignal/OneSignal-Website-SDK/blob/master/src/managers/ServiceWorkerManager.ts#L420) may we do it this way?

```
.catch(e => console.error("Failed to install Service Worker: " + e.message))
```

Or maybe an equivalent and standard way of logging within angular?  Or we'll have an `@Optional()` injectable, in example `onServiceWorkerErrorHandler` to specifically handle it.

I tried to handle it myself, but the following methods do not trigger, hence, i can't `.preventDefault()` on them:

```
# main.ts

(navigator.serviceWorker as any).onerror = () => console.log('111');
navigator.serviceWorker.onmessageerror = () => console.log('222');
navigator.serviceWorker.addEventListener('messageerror', () => console.log('333'));
```

The error does get handled by `ErrorHandler`, but there is no way for me distinguish if it really came from  `ngswAppInitializer` when i do:

```
@Injectable()
export class SentryErrorHandler extends ErrorHandler {
  handleError(error): void {
    ...
  }
}
```

Thanks @manklu , looking at https://alligator.io/ they handle it like so:

```
  navigator.serviceWorker.register('/sw3.js').catch(function(err) {
    console.log('ServiceWorker failed 😢', err);
  });
```

**Comment by gkalpak** [comment]
Hm...I don't get it. Why should the error not be passed to the default `ErrorHandler`?

**Comment by alxhub** [comment]
We should at least validate that the errors are routed to the right place.

**Comment by angelkenneth** [comment]
Seems to have been fixed by #30876 and released for Angular 8.1.0 , after upgrading to the latest Angular version (8.1.1): i no longer see the issue in Sentry. Closing.
